client.go

```diff
@@ -593,6 +593,16 @@ func (c *Client) Truncate(path string, size int64) error {
 	return c.setstat(path, sshFileXferAttrSize, uint64(size))
 }
 
+// SetExtendedAttrs sets the extended attributes of the named file.
+func (c *Client) SetExtendedAttrs(path string, extended []StatExtended) error {
+	type extendedAttrs struct {
+		Size     uint32
+		Extended []StatExtended
+	}
+	attrs := extendedAttrs{Size: uint32(len(extended)), Extended: extended}
+	return c.setstat(path, sshFileXferAttrExtended, attrs)
```

Comment:
Please use `&FileStat{ Extended: extended }` instead of a custom `struct`.

Reply:
This was a very old commit locally, it seems that I followed the example of `Lchown` and others above.

Reply:
I think I changed the `Chown` function recently as well. 😂 It used to be just like `[]uint32{ uid, gid}` or something.